src/index.ts

```diff
@@ -132,16 +226,109 @@ function createRdfEmitter(program: Program) {
         result
       );
     });
+
+
+  }
+
+  function checkIfDataProperty(model: Model) {
+    const intrinsic = getIntrinsicModelName(program, model);
+
+    if (!intrinsic) {
+      return false;
+    }
+    else{
+      return true;
+    }
   }
 
   function nameForModel(model: Model) {
     const intrinsic = getIntrinsicModelName(program, model);
 
     if (!intrinsic) {
+
       let ns = getNsForModel(model);
-      return ns.prefix + ":" + model.name;
+
+      if (model.name == "Array")
+      {
+        if (model.templateArguments != undefined)
+        {
+          return ns.prefix + ":" + ((<any>model.templateArguments[0]).name);
+        }
+      }
+      else{
+        return ns.prefix + ":" + model.name;
+      }
     }
 
+
+    switch (intrinsic) {
+      case "boolean":
+        return "xsd:boolean";
+      case "bytes":
+        return "xsd:hexBinary"; // could be base64, check format?
+      case "duration":
+        return "xsd:dayTimeDuration";
+      case "float":
+        return "xsd:double"; // seems best for either float32 or float64?
+      case "float32":
+        return "xsd:float";
+      case "float64":
+        return "xsd:double";
+      case "integer":
+        return "xsd:integer";
+      case "uint8":
+        return "xsd:unsignedByte";
+      case "uint16":
+        return "xsd:unsignedShort";
+      case "uint32":
+        return "xsd:unsignedIntS";
+      case "uint64":
+        return "xsd:unsignedLong";
+      case "int8":
+        return "xsd:byte";
+      case "int16":
+        return "xsd:short";
+      case "int32":
+        return "xsd:int";
+      case "int64":
+        return "xsd:long";
+      case "safeint":
+        return "xsd:long";
+      case "string":
+        return "xsd:string";
+      case "numeric":
+        return "xsd:decimal";
+      case "plainDate":
+        return "xsd:date";
+      case "plainTime":
+        return "xsd:time";
+      case "zonedDateTime":
+        return "xsd:dateTime";
+      default:
+        throw new Error("xsd datatype not defined for instrinsic " + intrinsic);
+    }
+  }
+
+  function nameForModelSHACL(model: Model) {
+    const intrinsic = getIntrinsicModelName(program, model);

```

Comment:
This code seems identical to the above name except the latter has `_NodeShape` appended? If so, could this be just `return nameForModel(model) + "_NodeShape"`?